stellargraph/mapper/mini_batch_node_generators.py

```diff
@@ -341,7 +341,8 @@ def __getitem__(self, index):
             # Dictionary to store node indices for quicker node index lookups
             # The list of indices of the target nodes in self.node_list
             cluster_target_indices = np.array(
-                [self.target_node_lookup[n] for n in target_nodes_in_cluster]
+                [self.target_node_lookup[n] for n in target_nodes_in_cluster],
+                dtype=np.int64,
```

Comment:
Why is this bug only revealed with this patch?

Reply:
because the clusters returned by Louvain were pretty small and so some didn;t have any target nodes. I guess its just chance that this has never happened before + i think we've only every run cluster gcn on large clusters (relative graph size).

Reply:
Ah makes sense. Maybe this could be simultaneously changed in a separate PR that adds a test (with the thinking being that PR would land first, but also that it doesn't particularly matter if it lands second)?

Reply:
https://github.com/stellargraph/stellargraph/pull/1622